### PR TITLE
[Storage] Upload Blob From Url.

### PR DESCRIPTION
**Pull request by kasobol-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    /**
+     * @return {@link BlobHttpHeaders}
+     */
+    public BlobHttpHeaders getHeaders() {
+        return headers;
+    }
+
+    /**
+     * @param headers {@link BlobHttpHeaders}
+     * @return The updated {@code AppendBlobCreateOptions}
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java:52]: BlobUploadFromUrlOptions

Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java:52]: Or just options.

**Review by rickle-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+import com.azure.storage.blob.models.BlobHttpHeaders;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.common.implementation.StorageImplUtils;
+
+import java.util.Map;
+
+/**
+ * Extended options that may be passed when uploading a Block Blob from URL.
+ */
+public class BlobUploadFromUrlOptions {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java:17]: If this API is on block blob, should the options be BlockBlobUploadFromUrlOptions?

Comment (kasobol-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java:17]: @gapra-msft The short term decision is to put this on Block Blob only to let it mature, but long term goal is to put this on BlobClient as well and advertise as first choice over copy.

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java:17]: Cool. Sounds good to me then

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**